### PR TITLE
PCHR-478: Create New Assignment Modal Bulk Assign

### PR DESCRIPTION
**Pull request by kkalamarski**
Bulk Assign feature + Changes in layout to reflect newest design.

Before:
before-478

After:
after-478

Copy button:
bulk-assign

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/js/app.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ define([
     'services/services',
 ], function(){
     'use strict';
-
+
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/app.js]: Unnecessary change?

Comment (kkalamarski) [inline uk.co.compucorp.civicrm.tasksassignments/js/app.js]: Yes, probably because of code reformat.

File: uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalAssignment.js [hunk 2]
```diff
@@ -217,7 +216,15 @@ define(['controllers/controllers',
 
             };
 
-            $scope.$watch('activitySet',function(activitySet){
+            $scope.setAssignee = function setAssignee(list) {
+                angular.forEach(list, function (item) {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalAssignment.js]: It's safe enough i think to just use the built-in ECMAScript 5's `forEach`

File: uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalAssignment.js [hunk 3]
```diff
@@ -245,7 +252,7 @@ define(['controllers/controllers',
                         continue;
                     }
 
-                    taskType = activity.name ? $filter('filter')($rootScope.cache.taskType.arr, { value: activity.name }, true)[0] : '';
+                    taskType = activity.name ? $filter('filter')($rootScope.cache.taskType.arr, {value: activity.name}, true)[0] : '';
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalAssignment.js]: I believe it's better to have a padding whitespace in object literals, like it was before

File: uk.co.compucorp.civicrm.tasksassignments/js/controllers/task.js [hunk 4]
```diff
@@ -1,32 +1,32 @@
...
+    'moment',
+    'services/task'
+], function (controllers, moment) {
+    controllers.controller('TaskCtrl', ['$scope', '$log', '$rootScope',
+        function ($scope, $log, $rootScope) {
             $log.debug('Controller: TaskCtrl');
 
             $scope.isCollapsed = true;
-            $scope.picker = { opened: false };
+            $scope.picker = {opened: false};
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/controllers/task.js]: Same as above, and same for any following instance

File: uk.co.compucorp.civicrm.tasksassignments/sass/civihr/partials/_modals.scss [hunk 5]
```diff
@@ -54,6 +54,15 @@
   }
 }
 
+.modal-inverted{
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/sass/civihr/partials/_modals.scss]: Always leave a space between the selector and the open bracket
```css
.modal-inverted {
```

File: uk.co.compucorp.civicrm.tasksassignments/sass/civihr/partials/_modals.scss [hunk 6]
```diff
@@ -54,6 +54,15 @@
...
+  .modal-header, .modal-footer{
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/sass/civihr/partials/_modals.scss]: Stack multiple selectors vertically
```css
.modal-header,
.modal-footer {
```

File: uk.co.compucorp.civicrm.tasksassignments/sass/civihr/partials/_modals.scss [hunk 7]
```diff
@@ -54,6 +54,15 @@
...
+    background-color: $brand-secondary;
+  }
+  .modal-body{
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/sass/civihr/partials/_modals.scss]: Always put a new line between one css rule and another
```css
.modal-header,
.modal-footer {
    /* ... */
}

.modal-body {
  /* ... */
}

File: uk.co.compucorp.civicrm.tasksassignments/sass/civitasks/partials/_forms.scss [hunk 8]
```diff
@@ -8,6 +8,96 @@
...
 
+.form-assignment {
+  h2 {
+    font-size: 16px;
+    line-height: 32px;
+    margin: 0;
+  }
+
+  hr {
+    margin: 0 -15px 10px -15px;
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/sass/civitasks/partials/_forms.scss]:
```css
hr {
    margin: 0 -15px 10px;
}
```

File: uk.co.compucorp.civicrm.tasksassignments/sass/civitasks/partials/_forms.scss [hunk 9]
```diff
@@ -8,6 +8,96 @@
...
+  }
+
+  .row {
+    margin-bottom: 10px;
+  }
+
+  .display-only {
+    &.header {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/sass/civitasks/partials/_forms.scss]: Leave a new line between the parent and nested selector
```css
.display-only {

    &.header {
        /* ... */
    }
}

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/assignment.html [hunk 10]
```diff
@@ -1,250 +1,332 @@
-<form novalidate name="assignmentForm" class="form-assignment form-horizontal" role="form" ng-submit="confirm()" ct-spinner>
+<form novalidate name="assignmentForm" class="form-assignment form-horizontal modal-inverted" role="form"
+      ng-submit="confirm()"
+      ct-spinner>
     <div class="modal-header">
-        <button type="button" class="close" ng-click="cancel()"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
-        <h4 class="modal-title" ng-bind="cache.contact.obj[assignment.contact_id].sort_name || 'No target contact'"></h4>
+        <button type="button" class="close" ng-click="cancel()"><span aria-hidden="true">&times;</span><span
+                class="sr-only">Close</span></button>
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/assignment.html]: *(Applies for any following instance)*

If you're gonna split the line on multiple lines, might as well then do the split consistent with the markup structure

```html
<button type="button" class="close" ng-click="cancel()">
    <span aria-hidden="true">&times;</span>
    <span class="sr-only">Close</span>
</button>
```

**Comment by AkA84** [comment]
Minor comments, once address it's approved
